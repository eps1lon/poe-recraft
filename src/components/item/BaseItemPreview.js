// @flow
import classnames from 'classnames';
import { type Item } from 'poe-mods';
import React from 'react';

import './assets/item.css';

import Header from './Header';
import Content from './Content';

export type Props = {
  item: Item
};

const BaseItemPreview = ({ item }: Props) => {
  return (
    <div className={classnames('itembox', item.rarity.toString())}>
      <Header item={item} />
      <Content item={item} />
    </div>
  );
};

export default BaseItemPreview;
