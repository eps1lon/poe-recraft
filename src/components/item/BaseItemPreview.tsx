import classnames from 'classnames';
import { Item } from 'poe-mods';
import React, { SFC } from 'react';

import './assets/item.css';

import Header from './Header';
import Content from './Content';

export type Props = {
  item: Item;
};

const BaseItemPreview: SFC<Props> = ({ item }) => {
  return (
    <div className={classnames('itembox', item.rarity.toString())}>
      <Header item={item} />
      <Content item={item} />
    </div>
  );
};

export default BaseItemPreview;
