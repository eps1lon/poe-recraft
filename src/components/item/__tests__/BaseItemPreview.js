import { createItems } from 'poe-mods';
import React from 'react';
import renderer from 'react-test-renderer';

import BaseItemPreview from '../BaseItemPreview';

const items = createItems(
  require('../../../../public/data/baseitemtypes.json')
);

it.skip('should render boots', () => {
  const boots = items.from(({ name }) => name === 'Iron Greaves');

  const component = renderer.create(<BaseItemPreview item={boots} />);

  expect(component.toJSON()).toMatchSnapshot();
});
