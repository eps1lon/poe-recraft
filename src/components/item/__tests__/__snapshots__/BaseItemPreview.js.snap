// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render boots 1`] = `
<div
  className="itembox normal"
>
  <div
    className="header"
  >
    <span
      className="name-left"
    />
    <div
      className="item-name typeline"
    >
      Iron Greaves
    </div>
    <span
      className="name-right"
    />
  </div>
  <div
    className="content"
  >
    <div
      className="display-property"
    >
      <span>
        Armour
      </span>
      :
       
      <span
        className="value simple"
      >
        6
      </span>
    </div>
    <div
      className="separator"
    />
    <div>
      Requires 
      <span>
        Level 
      </span>
      <span
        className="value"
      >
        1
      </span>
      , 
      <span
        className="value"
      >
        8
      </span>
      <span>
         Str
      </span>
    </div>
  </div>
</div>
`;
